
public class four {
	
	static void demo() {
		try {
			throw new NullPointerException("demo"); //NullPointerException is an object of type throw
		}
		catch(NullPointerException e){
			System.out.println("we are inside the catch ");
			throw e;
		}
	}
	
	public static void main(String[] args) {
		try {
			demo();
		}
			catch(NullPointerException e) {
				System.out.println("recaught: " + e);
			}
		
	}
}
